SpokeStack/SpokeStack/WakeWordSpeechRecognizer.swift

```diff
@@ -0,0 +1,660 @@
+
+                } catch SpeechPipelineError.illegalState(let message) {
+                    
+                    fatalError("Error occured while smoothing \(message)")
+                    
+                } catch {
+                    
+                     fatalError("Error occured while smoothing \(#line)")
+                }
+            }
+        }
+        
+        /// Compute the posterior mean of each keyword class
+        /// Write the outputs to the phrasing window
+        
+        let total: Int = self.smoothWindow.capacity / self.words.count
+        self.phraseWindow.rewind().seek(self.words.count)
+
+        for (index, _) in self.words.enumerated() {
+            
+            do {
+                
+                let windowValue: Float = self.phraseSum[index] / Float(total)
+                try self.phraseWindow.write(windowValue)
+                
+                print("windowValue \(windowValue)")
+
+            } catch SpeechPipelineError.illegalState(let message) {
+                
+                fatalError("illegal state error \(message)")
+                
+            } catch {
+                
+                fatalError("Error occured while smoothing \(#line)")
+            }
+        }
+        
+        self.phrase()
+    }
+    
+    private func phrase() -> Void {
+        
+        /// Compute the argmax (winning class) of each smoothed output
+        /// in the current phrase window
+        
+        var index: Int = 0
+        var max: Float = -Float.greatestFiniteMagnitude
+        
+        repeat {
+            
+            for (subindex, _) in self.words.enumerated() {
+
+                do {
+
+                    let value: Float = try self.phraseWindow.read()
+                    
+//                    print("value in the phrase \(value)")
+                    
+                    if value > max {
+                    
+                        self.phraseArg[index] = subindex
+                        max = value
+                    }
+
+                } catch SpeechPipelineError.illegalState(let message) {
+                    
+                    fatalError("illegal state error \(message)")
+                    
+                } catch {
+                    
+                    fatalError("Error occured while phrase \(#line)")
+                }
+            }
+            
+            index += 1
+
+        } while !self.phraseWindow.isEmpty
+        
+        /// Attempt to find a matching phrase amoung the argmaxes
+        
+        phrasesArgumentLabel: for phrase in self.phrases {
+            
+            /// Search for any occurrences of the phrase's keywords in order
+            /// across the whole phrase window
+            
+            var match: Int = 0
+            
+            phraseArgumentLabel: for word in self.phraseArg {
+                
+                if word == phrase[match] {
+                    
+                    match -= 1
```

Comment:
I believe this should be an increment: https://github.com/pylon/spokestack-android/blob/master/src/main/java/com/pylon/spokestack/wakeword/WakewordTrigger.java#L519